lineapy/execution/inspect_function.py

```diff
@@ -105,14 +110,33 @@ def get_specs() -> Dict[str, List[Annotation]]:
     relative_path = "../*.annotations.yaml"
     path = os.path.join(os.path.dirname(__file__), relative_path)
     valid_specs: Dict[str, List[Annotation]] = defaultdict(list)
+
+    def add_spec_if_valid(f):
+        doc = yaml.safe_load(f)
+        for item in doc:
+            v = validate(item)
+            if v is None:
+                continue
+            valid_specs[v.module].extend(v.annotations)
+
     for filename in glob.glob(path):
         with open(filename, "r") as f:
-            doc = yaml.safe_load(f)
-            for item in doc:
-                v = validate(item)
-                if v is None:
-                    continue
-                valid_specs[v.module].extend(v.annotations)
+            add_spec_if_valid(f)
+
+    # Ingest custom annotation specs in ~/.lineapy/custom-annotations/
+    annotate_folder = custom_annotations_folder()
+    for annotation_path in annotate_folder.iterdir():
+        # Skip if not .annotations.yaml file
+        if (
+            not annotation_path.is_file()
+            or re.match(CUSTOM_ANNOTATIONS_REGEX_MATCH, str(annotation_path))
+            is None
+        ):
+            continue
+
+        with annotation_path.open() as f:
+            add_spec_if_valid(f)
+
```

Comment:
two pts here: 
1 ) We are checking for valid annotations in two places, using cli > validate and here. My suggestion is to do leave the validation when running `lineapy annotate validate`. Alternately, we can also run validations when adding an annotation and only add valid files.

2) With validation out of the way, this whole piece can go away and be combined with the path variable initialized above like so:
```
paths = glob.glob(os.path.join(os.path.dirname(__file__), relative_path))
paths = paths.extend(os.path.join(custom_annotations_folder().resolve(), relative_path))
for filename in paths:
...
```
This will also take away the need for regex matches.